examples/advanced/vertical_xgboost/README.md

```diff
@@ -0,0 +1,81 @@
+```
+python3 -m pip install -r requirements.txt
+```
+
+## Preparing HIGGS Data
+In this example we showcase a binary classification task based on the [HIGGS dataset](https://archive.ics.uci.edu/dataset/280/higgs), which contains 11 million instances, each with 28 features and 1 class label.
+
+### Download and Store Dataset
+We first download the dataset from the HIGGS link above, which is a single zipped `.csv` file.
+By default, we assume the dataset is downloaded, uncompressed, and stored in `~/dataset/HIGGS.csv`.
+
+### Vertical Data Splits
+In vertical federated learning, sites share overlapping data samples (rows), but contain different features (columns).
+In order to achieve this, we split the HIGGS dataset both horizontally and vertically. As a result, each site has an overlapping subset of the rows and a  subset of the 29 columns.
+
+<img src="./figs/vertical_fl.png" alt="vertical fl diagram" width="500"/>
+
+Run the following commands to prepare the data splits and job configurations:
+```
+./prepare_data.sh
+./prepare_job_config.sh
+```
+
+### Private Set Intersection (PSI)
+Since not every site will have the same set of data samples, we can use PSI to compare encrypted versions of clients datasets in order to jointly compute the intersection based on common IDs. In this example, the HIGGS dataset does not contain unique identifiers so we add a temporary `uid_{idx}` to each instance and give each site a portion of the HIGGS dataset (rows) that includes a common overlap. Afterwards the identifiers are dropped since they are only used for matching, and training is then done on the intersected data. To learn more about our PSI protocol implementation, see our [psi example](../psi/README.md).
+
+> **_NOTE:_** The uid can be a composition of multiple variabes with a transformation, however in this example we use indices for simplicity. PSI can also be used for computing the intersection of overlapping features, but here we give each site unique features.
+
+## XGBoost Federated Learning Plugin with FLARE
+
+### Installation
+
+Since the XGBoost federated learning plugin is not available in the official XGBoost releases yet, we must build XGBoost with the plugin from the source code.
+First clone the repo:
+```
+git clone --recursive https://github.com/dmlc/xgboost
+```
+Next, follow the steps [here](https://github.com/dmlc/xgboost/blob/master/plugin/federated/README.md) to install gRPC, build XGBoost with the federated plugin enabled, and install the python package into your virtual environemnt.
+
+### Integration with FLARE
+The plugin allows the XGBoost internal training parameter `DataSplitMode` to be set to column split mode rather than row split mode for distributed training. Additionally, all communication is done through the federated communicator, such as broadcasting split results and gradients.
+
+Information about the federated communicator is defined in the XGBoost communicator environment:
```

Comment:
This is necessary to pass into `xgb.DMatrix(... , data_split_mode=1)` for column based splits. I can clarify the readme to focus on setting this parameter rather than mention the internals.